base/abstractarraymath.jl

```diff
@@ -2,8 +2,6 @@
 
  ## Basic functions ##
 
-isinteger(x::AbstractArray) = all(isinteger,x)
-isinteger{T<:Integer,n}(x::AbstractArray{T,n}) = true
```

Comment:
I think adding `@inbounds` to `all` is not safe unless we add a specialization for `Array` (where we know iteration never goes out of bounds).